packages/patternfly-4/react-core/src/components/Switch/Switch.tsx

```diff
@@ -70,7 +70,25 @@ class Switch extends React.Component<SwitchProps & InjectedOuiaProps> {
           disabled={isDisabled}
           aria-labelledby={isAriaLabelledBy ? `${this.id}-on` : null}
         />
-        {label !== '' || labelOff !== '' ? (
+        {label !== '' && labelOff === '' ? (
+          <React.Fragment>
+            <span className={css(styles.switchToggle)} />
+            <span
+              className={css(styles.switchLabel, styles.modifiers.on)}
+              id={isAriaLabelledBy ? `${this.id}-on` : null}
+              aria-hidden="true"
+            >
+              {label}
+            </span>
+            <span
+              className={css(styles.switchLabel, styles.modifiers.off)}
+              id={isAriaLabelledBy ? `${this.id}-off` : null}
+              aria-hidden="true"
+            >
+              {label}
```

Comment:
should remove labelOff from the conditional above and simplify this. 
```suggestion
              {labelOff || label}
```

Reply:
If I'm not wrong, in that case labelOff is equal to `''` (empty string) which will evaluate `label` value.